vocabularies/Common.json

```diff
@@ -1288,6 +1288,51 @@
       "@Common.Experimental": true,
       "@Core.IsURL": true,
       "@Core.Description": "Base URL for WebSocket connections"
+    },
+    "RecommendationsAction": {
+      "$Kind": "Term",
+      "$AppliesTo": ["EntityType"],
+      "@Common.Experimental": true,
+      "@Core.Description": "Name of an action (or function) import that retrieves AI-based recommendations (see [template](#Template_RecommendationsAction)) for an entity of the annotated type"
+    },
+    "Template_RecommendationsAction": [
+      {
+        "$Kind": "Action",
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for actions that retrieve AI-based recommendations for an entity and its related entities",
+        "@Core.LongDescription": "The template action itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "EntityKeys",
+            "$Type": "Edm.ComplexType",
+            "@Core.Description": "Nested structure containing the keys of entities for which recommendations shall be retrieved",
+            "@Core.LongDescription": "The nested complex type of this parameter has the same structure as the entity type bearing the [`GetRecommendationsAction`](#GetRecommendationsAction) annotation\n            and its related entity types, but contains only their key properties.\n            If this parameter can be included in a URL, a function can be used instead of an action."
```

Comment:
- GetRecommendationsAction should be changed to RecommendationsAction (text as well as reference
- Current approach increases the effort for the application developer by mandating the type definition for parameter. Is there a way this could be generated on the fly by the Framework

Reply:
> * Current approach increases the effort for the application developer by mandating the type definition for parameter.

We can try to reuse the entity type.